utils/utils.c

```diff
@@ -46,11 +46,13 @@ void xdag_init_path(char *path)
     
     strcpy(g_xdag_current_path, szBuffer);
 #else
-    char result[PATH_MAX];
-    if (readlink("/proc/self/exe", result, PATH_MAX) > 0) {
-		strcpy(g_xdag_current_path, result);
+    char * n = dirname(path);
```

Comment:
Could you please use more meaningful name for variable instead of "n"?
It looks like this section is the same as for Mac now, can you combine them?